pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4156,10 +4156,13 @@ def roll_qtrday(other: datetime, n: int, month: int,
     """
     cdef:
         int months_since
-    # TODO: Merge this with roll_yearday by setting modby=12 there?
-    #       code de-duplication versus perf hit?
     # TODO: with small adjustments this could be used in shift_quarters
-    months_since = other.month % modby - month % modby
+
+    if modby == 12:
```

Comment:
can you check if modby is 12 or 3 else error. should this be a required arg?

Reply:
can u do this here

Reply:
id rather not add a check here bc we're going to be calling this from within a (nogil) loop

Reply:
ok i think this should be a required arg then (then no check is needed); its too easy to accidently mess this up

Reply:
k, next branch